docs/docs/design/bitfield-signing.md

```diff
@@ -0,0 +1,90 @@
+# Bitfield Signing Subsystem
+
+[The bitfield signing subsystem](https://paritytech.github.io/polkadot-sdk/book/node/availability/bitfield-signing.html)
+is responsible for producing signed availability bitfields and passing them to
+[the bitfield distribution subsystem](./bitfield-distribution.md). It only does that when running on a validator.
+
+## Subsystem Structure
+
+The implementation must conform to the `Subsystem` interface defined in the `parachaintypes` package. It should live in
+a package named `bitfieldsigning` under `dot/parachain/bitfield-signing`.
+
+### Messages Received
+
+The subsystem must be registered with the overseer. It only handles `parachaintypes.ActiveLeavesUpdateSignal`. The
+overseer must be modified to forward this message to the subsystem.
+
+### Messages Sent
+
+1. [`DistributeBitfield`](https://github.com/paritytech/polkadot-sdk/blob/1e3b8e1639c1cf784eabf0a9afcab1f3987e0ca4/polkadot/node/subsystem-types/src/messages.rs#L522)
+
+```go
+package parachaintypes
+
+import "github.com/ChainSafe/gossamer/lib/common"
+
+type DistributeBitfield struct {
+	RelayParent common.Hash
+	Bitfield    UncheckedSignedAvailabilityBitfield
+}
+```
+
+As mentioned in [bitfield distribution](./bitfield-distribution.md), it makes sense to duplicate the type
+`UncheckedSignedAvailabilityBitfield` as `CheckedSignedAvailabilityBitfield` or just `SignedAvailabilityBitfield` and
+use it in `DistributeBitfield`.
+
+2. [`AvailabilityStore::QueryChunkAvailability(CandidateHash, ValidatorIndex, response_channel)`](https://github.com/paritytech/polkadot-sdk/blob/1e3b8e1639c1cf784eabf0a9afcab1f3987e0ca4/polkadot/node/subsystem-types/src/messages.rs#L556)
```

Comment:
could you describe what leads this message to be sent?

Reply:
done